compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/Common.cpp

```diff
@@ -770,7 +770,7 @@ static LogicalResult matchAndSetReductionStrategy(func::FuncOp entryPoint,
       StagedReductionStrategy strategy(captures, reductionConfig);
       return buildStagedReductionStrategy(b, variant, strategy);
     } else {
-      return llvm_unreachable("Unknown strategy");
+      llvm_unreachable("Unknown strategy");
```

Comment:
This looks like it is actually an exhaustive enum switch, where I think crashing is reasonable (but undefined behavior is not). Enum is defined here:

https://github.com/openxla/iree/blob/df71589e19fe6de33cdd03577f5ce807d0d129cd/compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/Common.h#L199

Is there some reason this isn't actually a switch instead of an if-else chain?

Reply:
Sorry, I don't know much about this code. I just fixed a bug that was exposed with the macro change (returning `llvm_unreachable` is incorrect). Happy to change it to whatever is needed.